src/Workspaces/Core/Portable/Workspace/Host/Mef/WorkspaceKindMetadata.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Host.Mef
+{
+    /// <summary>
+    /// MEF metadata class used to find exports declared for a specific <see cref="WorkspaceKind"/>.
+    /// </summary>
+    internal class WorkspaceKindMetadata
+    {
+        public string[] WorkspaceKinds { get; }
+
+        public WorkspaceKindMetadata(IDictionary<string, object> data)
+        {
+            this.WorkspaceKinds = (string[])data.GetValueOrDefault("WorkspaceKinds");
```

Comment:
nameof? (Although I wish you could write `nameof(ExportWorkspaceEventListenerAttribute.WorkspaceKinds)` but I don't think you can?)

Reply:
sure.